File: backend/src/application/worker_service.py
from src.domain.worker.worker_repository import WorkerRepository
from src.domain.worker.worker import Worker, WorkerCreate, WorkerUpdate
from typing import List, Optional

class WorkerService:
    def __init__(self, worker_repository: WorkerRepository):
        self.worker_repository = worker_repository

    def create_worker(self, worker: WorkerCreate) -> Worker:
        return self.worker_repository.create(worker)

    def get_all_workers(self) -> List[Worker]:
        return self.worker_repository.get_all()

    def get_worker_by_id(self, worker_id: str) -> Optional[Worker]:
        return self.worker_repository.get_by_id(worker_id)

    def update_worker(self, worker_id: str, worker_update: WorkerUpdate) -> Optional[Worker]:
        return self.worker_repository.update(worker_id, worker_update)

    def delete_worker(self, worker_id: str) -> bool:
        return self.worker_repository.delete(worker_id)

File: golang/devops_console/internal/application/worker_service.go
package application

import (
    "devops_console/internal/domain/worker"
)

type WorkerService struct {
    workerRepository worker.WorkerRepository
}

func NewWorkerService(workerRepository worker.WorkerRepository) *WorkerService {
    return &WorkerService{workerRepository: workerRepository}
}

func (s *WorkerService) CreateWorker(workerCreate worker.WorkerCreate) (*worker.Worker, error) {
    return s.workerRepository.Create(workerCreate)
}

func (s *WorkerService) GetAllWorkers() ([]*worker.Worker, error) {
    return s.workerRepository.GetAll()
}

func (s *WorkerService) GetWorkerByID(workerID string) (*worker.Worker, error) {
    return s.workerRepository.GetByID(workerID)
}

func (s *WorkerService) UpdateWorker(workerID string, workerUpdate worker.WorkerUpdate) (*worker.Worker, error) {
    return s.workerRepository.Update(workerID, workerUpdate)
}

func (s *WorkerService) DeleteWorker(workerID string) error {
    return s.workerRepository.Delete(workerID)
}